public/app/engagementindex/leaderboard/leaderboardController.js

```diff
@@ -83,128 +86,137 @@
         points.push($scope.users[i].points);
       }
 
-      // Wait until Angular has finished rendering the box plot container on the screen
-      setTimeout(function() {
-        // Render the box plot using highcharts
-        // @see http://api.highcharts.com/highcharts
-        var chart = new Highcharts.Chart({
-          'chart': {
-            'backgroundColor': 'transparent',
-            'inverted': true,
-            // Ensure that the box plot is displayed horizontally
-            'margin': [0, 20, 0, 20],
-            'renderTo': 'leaderboard-userinfo-boxplot',
-            'type': 'boxplot'
-          },
-
-          'title':{
-            // Ensure that no chart title is rendered
-            'text':''
-          },
-
-          'legend': {
-            // Ensure that no legend is rendered
-            'enabled': false
-          },
-
-          'credits': {
-            // Ensure that no highcarts watermark is rendered
-            'enabled': false
-          },
-
-          'tooltip': {
-            'hideDelay': 100,
-            'positioner': function(labelWidth, labelHeight) {
-              // Ensure that the tooltip does not overlap with the box plot to
-              // allow access hover access to 'my points'
-              return {
-                x: 305,
-                y: 15 - (labelHeight / 2)
-              };
+      // Calculate the boxplot data (min, q1, q2, q3, max)
+      var boxplotData = calculateBoxPlotData(points);
+
+      // Only render the boxplot when there's enough data
+      if (_.uniq(boxplotData).length >= 2) {
+
+        $scope.showBoxplot = true;
+
+        // Wait until Angular has finished rendering the box plot container on the screen
+        setTimeout(function() {
+          // Render the box plot using highcharts
+          // @see http://api.highcharts.com/highcharts
+          var chart = new Highcharts.Chart({
+            'chart': {
+              'backgroundColor': 'transparent',
+              'inverted': true,
+              // Ensure that the box plot is displayed horizontally
+              'margin': [0, 20, 0, 20],
+              'renderTo': 'leaderboard-userinfo-boxplot',
+              'type': 'boxplot'
             },
-            'shadow': false,
-            'style': {
-              'color': '#FFF'
+
+            'title':{
```

Comment:
@nicolaasmatthijs How nit-picky are you on style?  If "very" then space needed above.